Gorry's review (#112)
ietf-wg-scone/scone: draft-ietf-scone-protocol.md

in any way that is separate from any signals
that are intended to influence congestion response.

In addition to rate limiting policies,
throughput advice can indicate temporary increases in available capacity
or temporarily reduced capacity.
This includes persistent overuse, equipment faults, or other transient issues.
Providing advice is applicable if increases or reductions

> are expected to last at least one minute.

Author: I think that the problem is "one", not that the text is vague.  Because what we really want to say is "at least several monitoring periods".  There's no value in signaling for anything of shorter duration.

```suggestion
are expected to last for more than one monitoring period; see {{time}}.
```

The same UDP address tuple might be used for multiple QUIC connections.  A
single signal might be lost or only reach a single application endpoint.
Network elements can apply SCONE advice

> to all flows that include SCONE packets

Reviewer: ```suggestion
to all datagrams that include SCONE packets
```

Is this the intent?

Author: "flows" is correct in this context.  The title of the section is "Per-Flow Signal".

flow control credit (see {{Section 4 of QUIC}}) to indirectly limit the sending
rate of a peer.

Some applications offer options for rate control that can offer superior outcomes.
Applications that are able to adapt bit rates,
such as real-time and streaming video applications,
can adjust video quality to fit within a target throughput.
For instance, an HTTP Live Streaming {{?HLS=RFC8216}} or DASH {{DASH}}

> client can ask for lower bitrate, lower quality media segments.

Reviewer: Thanks, but as to your points:
* Regarding retransmissions, I'm not sure if it is relevant, as the problem exists with video bit rate selection. To paraphrase, SCONE assumes a low PLR.
* I wouldn't mind keeping status quo as-is, but if we want to explain why, maybe we should be more direct about the types of traffic might respect SCONE using flow control. We could use background software updates as an example, pointing out that, in some cases, avoiding CC-throttled is more important than the speed of background operation.

> client can ask for lower bitrate, lower quality media segments.

Reviewer: Opened #113 so that we could discuss the point separately.

flow control credit (see {{Section 4 of QUIC}}) to indirectly limit the sending
rate of a peer.

Some applications offer options for rate control that can offer superior outcomes.
Applications that are able to adapt bit rates,
such as real-time and streaming video applications,

> can adjust video quality to fit within a target throughput.

Reviewer 2: Seems slightly clunky. It could be something alonmg the lines of: "Video-based applications that can adjust video quality, such as real-time and streaming video applications,
can adjust their rate to fit within a target throughput"

Once the throughput advice signal is updated,
the network element updates the UDP checksum for the datagram.

To avoid throughput advice expiring,
a network element needs to ensure that it sends updated rate signals
with no more than a monitoring period ({{time}}) between each update.
Because this depends on the availability of SCONE packets
and packet loss can cause signals to be missed,
network elements might need to update more often.
Ideally, network elements update advice in SCONE packets

> twice per monitoring period,

Reviewer: ```suggestion
at least twice per monitoring period,
```
More than twice is even better.
